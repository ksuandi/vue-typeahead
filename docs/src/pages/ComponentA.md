# Component A

The description for Component A goes here.

### Example

```html
<template>
  <component-a :msg="msg"/>
</template>
<script>
export default {
  data () {
    return {
      msg: 'Greeting'
    }
  }
}
</script>
<!-- component-demo.vue -->
````

### Props

Name                 | Type       | Default      | Required | Description
----------------     | ---------- | ------------ | -------- | -----------------------
`msg`                | String     | Hello World! |          | A message for Component A
