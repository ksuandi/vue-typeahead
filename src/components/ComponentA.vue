<template>
  <div class="component-a">
    Value of 'msg' prop: <span class="message">{{msg}}</span>
  </div>
</template>

<script>
export default {
  props: {
    msg: {
      type: String,
      required: 'Hello World!'
    }
  }
}
</script>

<style lang="scss" scoped>
.component-a {
  text-align: center;
  padding: 10px;
  border: 1px solid lightgrey;
  border-radius: 5px;

  .message {
    color: #42b983;
  }
}
</style>
