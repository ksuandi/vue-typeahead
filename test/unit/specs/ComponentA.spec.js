import { mount } from '@vue/test-utils'
import ComponentA from '@/components/ComponentA'

const createCmp = propsData => mount(ComponentA, { propsData })

describe('ComponentA.vue', () => {
  let wrapper

  it('has a msg property', () => {
    wrapper = createCmp({ msg: 'hey' })
    expect(wrapper.props('msg', 'hey')).toBeTruthy()
  })

  it('has the expected html structure', () => {
    wrapper = createCmp({ msg: 'Greeting' })
    expect(wrapper.element).toMatchSnapshot()
  })
})
