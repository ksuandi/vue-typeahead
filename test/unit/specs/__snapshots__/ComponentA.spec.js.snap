// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ComponentA.vue has the expected html structure 1`] = `
<div
  class="component-a"
>
  
  Value of 'msg' prop: 
  <span
    class="message"
  >
    Greeting
  </span>
</div>
`;
